src/modules/order/components/DiscountContainer/DiscountContainer.tsx

```diff
@@ -1,23 +1,20 @@
 /** @jsxImportSource @emotion/react */
-import { discounts } from 'data';
 import React, { useState } from 'react';
+import { useSetRecoilState } from 'recoil';
 import styles from './DiscountContainer.styles';
+import { discounts } from 'data';
+import { pizzaAtoms } from 'modules/order/state/atoms';
 
-interface DiscountContainerProps {
-  onDiscount: (percentage: number) => void;
-}
-
-export const DiscountContainer: React.FC<DiscountContainerProps> = ({
-  onDiscount,
-}) => {
+export const DiscountContainer: React.FC = () => {
   const [input, setInput] = useState('');
+  const setDiscount = useSetRecoilState(pizzaAtoms.discount);
 
   function applyCode() {
     const discount = discounts.find((discount) => discount.code === input);
```

Comment:
In the future you if you have bigger objects you can destructure this.
`    const discount = discounts.find(({code}) => code === input);
`

Reply:
Also, you can make a useDiscount hook where you handle all the logic, just extract the logic from this component into the hook and then just call the functions in this component.
`const {applyCode,input,setInput,setDiscount = useDiscount() }`